api/src/main/java/org/eclipse/microprofile/concurrent/ThreadContext.java

```diff
@@ -106,6 +107,24 @@
      */
     static final String TRANSACTION = "Transaction";
 
+    /**
+     * <p>Creates an <code>Executor</code>that runs tasks on the same thread from which
+     * <code>execute</code>is invoked but with context that is captured from the thread
+     * that invokes <code>withCurrentContext</code>.</p>
+     *
+     * <p>Example usage:</p>
+     * <pre>
+     * <code>Executor contextSnapshot = threadContext.withCurrentContext();
+     * ...
+     * // from another thread, or after thread context has changed,
+     * contextSnapshot.execute(() -> obj.doSomethingThatNeedsContext());
+     * contextSnapshot.execute(() -> doSomethingElseThatNeedsContext(x, y));
+     * </code></pre>
+     *
+     * @return an executor that wraps the <code>execute</code> method with context.
+     */
+    Executor withCurrentContext();
```

Comment:
ThreadContext.withCurrentContext() does not make much sense/

How about
ThreadContext.getExecutorWithCurrentContext()

Reply:
I disagree, I think that's too long a name and `withCurrentContext` is in line with the other wrapper functions already present here. They do exactly the same kind of thing too.

Reply:
I can see an object to `withCurrentContext` being that in other cases these methods accept a parameter that they wrap _with_ the current context, whereas in this case nothing is wrapped and instead a new executor instance is created.  Another idea for a name could be `currentContextExecutor` or just `contextualExecutor`.  Would either of these be acceptable to both of you?

Reply:
Yeah,that's fine.